app/models/miq_report/generator.rb

```diff
@@ -191,127 +191,141 @@ def generate_table(options = {})
 
   def _generate_table(options = {})
     return build_table_from_report(options) if db == self.class.name # Build table based on data from passed in report object
+    _generate_table_prep
+
+    results = if custom_results_method
+                generate_custom_method_results(options)
+              elsif performance
+                generate_performance_results(options)
+              elsif interval == 'daily' && db_klass <= MetricRollup
+                generate_daily_metric_rollup_results(options)
+              elsif interval
+                generate_interval_metric_results(options)
+              else
+                generate_basic_results(options)
+              end
 
-    klass = db.kind_of?(Class) ? db : Object.const_get(db)
-    interval = db_options.present? && db_options[:interval]
-    custom_results_method = (db_options && db_options[:rpt_type]) ? "build_results_for_report_#{db_options[:rpt_type]}" : nil
-
-    includes = get_include_for_find
-
-    load_custom_attributes
-
-    time_profile.tz ||= tz if time_profile # Default time zone in profile to report time zone
-    ext_options = {:tz => tz, :time_profile => time_profile}
-    # TODO: these columns need to be converted to real SQL columns
-    # only_cols = cols
-    self.extras ||= {}
+    if db_options && db_options[:long_term_averages] && results.first.kind_of?(MetricRollup)
```

Comment:
darn. would have been nice to move the `long_term_averages` stuff into one of the `generate_` methods.

Having said that, most of these changes are just refactoring and don't involve too much research. so that may be good for a followup